Sources/Sentry/SentryTracer.m

```diff
@@ -469,7 +483,18 @@ - (void)finishInternal
     }];
 
     @synchronized(_children) {
-        if (self.idleTimeout > 0.0 && _children.count == 0) {
+        if ([self hasIdleTimeout] && _children.count == 0) {
+            SENTRY_LOG_DEBUG(
+                @"Was waiting for timeout for UI event trace and no children to wait for.");
+            if ([self didAutomaticTraceExceedMaxDuration]) {
+                SENTRY_LOG_DEBUG(@"Transaction exceeded max duration.");
+                return;
+            }
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+            [SentryProfiler stopProfilingSpan:self.rootSpan];
+            [SentryProfiler linkTransaction:_transaction];
+#endif // SENTRY_TARGET_PROFILING_SUPPORTED
+            [_hub captureTransaction:_transaction withScope:_hub.scope];
```

Comment:
I observed automatic transactions for eg button cilcks early returning here and never making it to the later code in `finishInternal` that created the transaction and captured it in an envelope. Does this mean we were never sending any transactions for button clicks? This also wound up capturing transactions that exceeded the hardcoded limit for prewarming duration, so added a second early bailout for that here too.